pkg/testing/tools/artifacts_api.go

```diff
@@ -0,0 +1,246 @@
+)
+
+var ErrBadHTTPStatusCode = errors.New("bad http status code")
+
+type VersionList struct {
+	Versions  []string `json:"versions"`
+	Aliases   []string `json:"aliases"`
+	Manifests struct {
+		LastUpdateTime         string `json:"last-update-time"`
+		SecondsSinceLastUpdate int    `json:"seconds-since-last-update"`
+	} `json:"manifests"`
+}
+
+type VersionBuilds struct {
+	Builds    []string `json:"builds"`
+	Manifests struct {
+		LastUpdateTime         string `json:"last-update-time"`
+		SecondsSinceLastUpdate int    `json:"seconds-since-last-update"`
+	} `json:"manifests"`
+}
+
+type Package struct {
+	URL        string `json:"url"`
+	ShaURL     string `json:"sha_url"`
+	AscURL     string `json:"asc_url"`
+	Type       string `json:"type"`
+	Attributes struct {
+		ArtifactNoKpi string `json:"artifactNoKpi"`
+		Internal      string `json:"internal"`
+		ArtifactID    string `json:"artifact_id"`
+		Oss           string `json:"oss"`
+		Group         string `json:"group"`
+	} `json:"attributes"`
+}
+
+type Project struct {
+	Branch                       string             `json:"branch"`
+	CommitHash                   string             `json:"commit_hash"`
+	CommitURL                    string             `json:"commit_url"`
+	ExternalArtifactsManifestURL string             `json:"external_artifacts_manifest_url"`
+	BuildDurationSeconds         int                `json:"build_duration_seconds"`
+	Packages                     map[string]Package `json:"packages"`
+	Dependencies                 []any              `json:"dependencies"`
+}
+
+type BuildDetails struct {
+	Build struct {
+		Projects             map[string]Project `json:"projects"`
+		StartTime            string             `json:"start_time"`
+		ReleaseBranch        string             `json:"release_branch"`
+		Prefix               string             `json:"prefix"`
+		EndTime              string             `json:"end_time"`
+		ManifestVersion      string             `json:"manifest_version"`
+		Version              string             `json:"version"`
+		Branch               string             `json:"branch"`
+		BuildID              string             `json:"build_id"`
+		BuildDurationSeconds int                `json:"build_duration_seconds"`
+	} `json:"build"`
+	Manifests struct {
+		LastUpdateTime         string `json:"last-update-time"`
+		SecondsSinceLastUpdate int    `json:"seconds-since-last-update"`
+	} `json:"manifests"`
+}
+
+type SearchPackageResult struct {
+	Packages  map[string]Package `json:"packages"`
+	Manifests struct {
+		LastUpdateTime         string `json:"last-update-time"`
+		SecondsSinceLastUpdate int    `json:"seconds-since-last-update"`
+	} `json:"manifests"`
+}
+
+type httpDoer interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+type ArtifactAPIClient struct {
+	c      httpDoer
+	apiURL string
+}
+
+func (aac ArtifactAPIClient) GetVersions(ctx context.Context) (list *VersionList, err error) {
```

Comment:
[Suggestion]
I'm not a heavy user of the API, but if you're returning a `VersionList` that is much more than just versions, I'd call it `GetVersionList` 
```suggestion
func (aac ArtifactAPIClient) GetVersionList(ctx context.Context) (list *VersionList, err error) {
```